klippy/extras/bulk_sensor.py

```diff
@@ -209,7 +209,7 @@ def __init__(self, clock_sync, bytes_per_sample):
         self.last_sequence = self.max_query_duration = 0
         self.last_overflows = 0
         self.mcu = self.oid = self.query_status_cmd = None
-    def setup_query_command(self, mcu, msgformat, oid, cq):
+    def setup_query_command(self, mcu, msgformat, oid, cq=None):
```

Comment:
It does not make sense to use the "default cmdqueue" for any of the bulk sensors.  So, this change does not seem correct.

For hx71x.py, just create a cmdqueue with `mcu.alloc_command_queue()`.

Reply:
Sure I can do that.

This is very non-obvious. The implementations down the stack don't allocate a command queue if one isn't provided, so I figured this was just an oversight. Maybe you should accept `None` here and do the allocation here because `bulk_sensor` has this special requirement?

Reply:
FYI, there isn't a special requirement for bulk sensors.  We want all devices to have their own queue so that messages to those devices are ordered wrt themselves, but not strictly ordered wrt other devices.  This allows the message priority system to prioritize urgent messages without corrupting ordering of a message stream to a device.

The low-level code can't (easily) assign command queues on its own, as it needs to know all the commands that could be sent to that device.  That is, all of "query_adxl345", "query_adxl345_status", "spi_send", and "spi_transfer" sent to a single adxl345 need to have strict ordering, and no code other than adxl345.py knows that.